tiled/adapters/jpeg.py

```diff
@@ -0,0 +1,279 @@
+import builtins
+from pathlib import Path
+from typing import Any, List, Optional, Tuple
+
+import numpy as np
+from numpy._typing import NDArray
+from PIL import Image
+
+from ..structures.array import ArrayStructure, BuiltinDtype
+from ..structures.core import Spec, StructureFamily
+from ..utils import path_from_uri
+from .protocols import AccessPolicy
+from .resource_cache import with_resource_cache
+from .type_alliases import JSON, NDSlice
+
+
+class JPEGAdapter:
```

Comment:
I wonder if there's a chance to define base classes, something like `ImageAdapter` and `ImageSequenceAdapter` and then subclass them for jpeg and tiff? What we will need to specify when subclassing are the supported mimetypes, and file reading functions (for a single file and for the sequence) -- otherwise the code looks almost identical.
Happy to help with restructuring it!

Reply:
Yes, that was what I thought too, first; for tiffs and jpegs alone, just passing a callable should be enough, but this may not extend to any other formats (or if there's need for some custom logic) in the future (this is partly why I was asking about pngs). I think, I'd be in favor of keeping seperate `JPEGAdapter` and `TiffAdapter` classes, even if they are very short.